SDK-2825: Rework the documentation for evaluator (#1931)
spryker/spryker-docs: docs/scos/dev/guidelines/keeping-a-project-upgradable/upgradability-guidelines/php-version.md

## Problem description

The PHP version is declared in different configuration files and used by dependencies.
You need to be sure that all those items have a PHP versions that are consistent and supported by [Spryker SDK](https://docs.spryker.com/docs/sdk/dev/spryker-sdk.html).

PHP versions are checked in:
- `composer.json`
- Config deploy files `deploy.**.yml`
- [Spryker SDK](https://docs.spryker.com/docs/sdk/dev/spryker-sdk.html) PHP versions

> ## Example of correct code

Reviewer: ```suggestion
## An example of correct code
```

> PHP version in deploy files should correspond the PHP version declared in `composer.json` and supported PHP version by [Spryker SDK](https://docs.spryker.com/docs/sdk/dev/spryker-sdk.html).

Reviewer: ```suggestion
The PHP version in your deploy files should correspond to the PHP version declared in `composer.json` and the supported PHP version by [Spryker SDK](https://docs.spryker.com/docs/sdk/dev/spryker-sdk.html).
```

...
```

> ## Example of evaluator error message

Reviewer: ```suggestion
## Example of an Evaluator error message
```

> Example 1. Unsupported [Spryker SDK](https://docs.spryker.com/docs/sdk/dev/spryker-sdk.html) PHP version is used in `composer.json` file.

Reviewer: ```suggestion
Below is an example of an unsupported [Spryker SDK](https://docs.spryker.com/docs/sdk/dev/spryker-sdk.html) PHP version being used in the `composer.json` file:
```

```shell
===================
PHP VERSION CHECKER
===================

+---+------------------------------------------------------------------------+---------------------------------+
| # | Message                                                                | Target                          |
+---+------------------------------------------------------------------------+---------------------------------+
| 1 | Composer json PHP constraint "7.2" does not match allowed PHP versions | <path_to_project>/composer.json |
+---+------------------------------------------------------------------------+---------------------------------+
```

> `composer.json` file, that produces the error message:

Reviewer: ```suggestion
A `composer.json` file that produces the error message:
```

```json
{
  "name": "spryker-shop/b2c-demo-shop",
  "description": "Spryker B2C Demo Shop",
  "license": "proprietary",
  "require": {
    "php": "7.2",
    ...
  }
}
```

> Example 2. Unsupported [Spryker SDK](https://docs.spryker.com/docs/sdk/dev/spryker-sdk.html) PHP version is used in `deploy.yml` file.

Reviewer: ```suggestion
Below is an example of an unsupported [Spryker SDK](https://docs.spryker.com/docs/sdk/dev/spryker-sdk.html) PHP version being used in the `deploy.yml` file.
```

```shell
===================
PHP VERSION CHECKER
===================

+---+-----------------------------------------------------------------------------------+------------------------------+
| # | Message                                                                           | Target                       |
+---+-----------------------------------------------------------------------------------+------------------------------+
| 1 | The deploy file uses a not allowed PHP image version "spryker/php:7.2-alpine3.12" | <path_to_project>/deploy.yml |
|   | The image tag must contain an allowed PHP version (image:abc-8.0)                 |                              |
+---+-----------------------------------------------------------------------------------+------------------------------+
```

> `deploy.yml` file, that produces the error message:

Reviewer: ```suggestion
The `deploy.yml` file that produces the error message:
```

> The `deploy.yml` file that produces the error message:

Reviewer: ```suggestion
A `deploy.yml` file that produces the error message:
```

```yaml
...
image:
    tag: spryker/php:7.2
    php:
        ini:
            "opcache.revalidate_freq": 0
            "opcache.validate_timestamps": 0
...
```

> Example 3. Inconsistent PHP versions are used in `composer.json` and `deploy.yml` files:

Reviewer: ```suggestion
Below is an example of inconsistent PHP versions being used in the `composer.json` and `deploy.yml` files:
```

```shell
===================
PHP VERSION CHECKER
===================

+---+--------------------------------------------+--------------------------------------------------------+
| # | Message                                    | Target                                                 |
+---+--------------------------------------------+--------------------------------------------------------+

> | 1 | Not all the targets have same PHP versions | Current php version $phpVersion: php7.2                |

Reviewer: ```suggestion
| 1 | Not all the targets have the same PHP versions | Current php version $phpVersion: php7.2                |
```

|   |                                            | tests/Acceptance/_data/InvalidProject/composer.json: - |
|   |                                            | tests/Acceptance/_data/InvalidProject/deploy**.yml: -  |
|   |                                            | SDK php versions: php7.2, php8.2                       |
+---+--------------------------------------------+--------------------------------------------------------+
```

> `composer.json` file uses PHP version `7.2`:

Reviewer: ```suggestion
The `composer.json` file uses PHP version `7.2`:
```

```json
{
  "name": "spryker-shop/b2c-demo-shop",
  "description": "Spryker B2C Demo Shop",
  "license": "proprietary",
  "require": {
    "php": "7.2",
    ...
  }
}
```

> `deploy.yml` file uses PHP version `8.2`:

Reviewer: ```suggestion
The `deploy.yml` file uses PHP version `8.2`:
```
